packages/integrations/vercel/src/serverless/adapter.ts

```diff
@@ -124,12 +138,25 @@ export default function vercelServerless({
 	return {
 		name: PACKAGE_NAME,
 		hooks: {
-			'astro:config:setup': ({ command, config, updateConfig, injectScript }) => {
-				if (command === 'build' && analytics) {
-					injectScript('page', 'import "@astrojs/vercel/analytics"');
+			'astro:config:setup': async ({ command, config, updateConfig, injectScript, logger }) => {
+				if (webAnalytics?.enabled || analytics) {
+					if (analytics) {
+						logger.warn(
+							`@astrojs/vercel: the \`analytics\` property is deprecated. Please use the new \`webAnalytics\` and \`speedInsights\` properties instead.`
```

Comment:
You don't need `@astrojs/vercel`. The logger will print the message with the name of the adapter already: https://docs.astro.build/en/reference/integrations-reference/#astrointegrationlogger